src/app/components/audio-recordings/pages/analysis-results/analyses-results.component.spec.ts

```diff
@@ -0,0 +1,212 @@
+import { AnalysisJobItemResult } from "@models/AnalysisJobItemResult";
+import { NgbTooltip } from "@ng-bootstrap/ng-bootstrap";
+import { createHostFactory, SpectatorHost } from "@ngneat/spectator";
+import { generateAnalysisJobResults } from "@test/fakes/AnalysisJobItemResult";
+import { MockProvider } from "ng-mocks";
+import { ToastrService } from "ngx-toastr";
+import { of } from "rxjs";
+import { AnalysesResultsComponent } from "./analyses-results.component";
+
+describe("analysesResultsComponent", () => {
+  let spectator: SpectatorHost<AnalysesResultsComponent>;
+
+  const createHost = createHostFactory({
+    declarations: [
+      NgbTooltip,
+      AnalysesDownloadRowComponent,
+      AnalysisDownloadWhitespaceComponent,
+    ],
+    component: AnalysesResultsComponent,
+    imports: [FormsModule],
+    mocks: [ToastrService],
+    providers: [
+      MockProvider(AnalysisJobItemResultsService),
+      MockProvider(ActivatedRoute),
+    ],
+  });
+
+  function setup() {
+    spectator = createHost("<baw-analyses-results></baw-analyses-results>");
+    spectator.detectChanges();
+  }
+
+  beforeEach(() => setup());
+
+  const getDirectoryRow = (index: number): HTMLElement =>
+    spectator.queryAll<HTMLElement>("baw-directory-row")[index];
+
+  const getDownloadButton = (index: number): HTMLButtonElement =>
+    getDirectoryRow(index).getElementsByTagName("button")[0];
+
+  const getDirectoryRowItemName = (index: number) =>
+    getDirectoryRow(index).innerText.split("\n")[0];
+
+  function clickFolder(index: number) {
+    spectator.queryAll<HTMLElement>(".grid-table-item")[3 + 3 * index].click();
+    updateComponent();
+  }
+
+  function updateComponent(): void {
+    spectator.component.ngOnChanges();
+    spectator.detectChanges();
+  }
+
+  function mockDirectoryStructure(newStructure: AnalysisJobItemResult[]) {
+    spectator.component.rows$ = of(newStructure);
+    updateComponent();
+
+    // as we are not interested in the root folder in many of our tests
+    // we need to assert that it exists, and open it up
+    const rootFolder = spectator.queryAll<HTMLElement>(".grid-table-item")[3];
+    expect(rootFolder).toBeTruthy();
+
+    rootFolder.click();
+    updateComponent();
+  }
+
+  it("should create", () => {
+    expect(spectator.component).toBeInstanceOf(AnalysesResultsComponent);
+  });
+
+  it("should display an enabled file download button for files", () => {
+    const singleFolderDirectory: AnalysisJobItemResult[] = [
+      new AnalysisJobItemResult(
+        generateAnalysisJobResults({
+          path: "/",
+          children: [
+            new AnalysisJobItemResult(
+              generateAnalysisJobResults({
+                path: "result_item.csv",
+              })
+            ),
+          ],
+        })
+      ),
+    ];
+
+    mockDirectoryStructure(singleFolderDirectory);
+
+    // get /FolderA and asset that it has the greyed out download zip button
```

Comment:
```suggestion
    // get /FolderA and assert that it has the greyed out download zip button
```

Further usages below to correct